src/__tests__/useFetch.test.tsx

```diff
@@ -449,22 +448,75 @@ describe('useFetch - BROWSER - with <Provider /> - Managed State', (): void => {
     expect(result.current.loading).toBe(false)
   })
 
-  it('should re-run the request when onUpdate dependencies are updated', async (done): Promise<void> => {
-    done()
-    let initialValue = 0
-    const { result, rerender, waitForNextUpdate } = renderHook(
-      () => useFetch({
-        data: {}
-      }, [initialValue]), // (onMount && onUpdate) === true
-      { wrapper }
+  it('should re-run the request when onUpdate dependencies are updated', async (): Promise<void> => {
+    const { result, waitForNextUpdate, rerender } = renderHook(
+      ({ initialValue }) =>
+        useFetch(
+          {
+            path: `/${initialValue}`,
+            data: {},
+          },
+          [initialValue],
+        ), // (onMount && onUpdate) === true
+      {
+        wrapper,
+        initialProps: {
+          initialValue: 0,
+        },
+      },
     )
+    // Default data
     expect(result.current.data).toEqual({})
-    initialValue = 1
-    rerender()
-    expect(result.current.loading).toEqual(true)
+
+    // Expected First Payload
+    await waitForNextUpdate()
+    expect(result.current.data).toEqual(expected)
+
+    // Expected Second Payload
+    rerender({
+      initialValue: 1,
+    })
+    await waitForNextUpdate()
+    expect(result.current.data).toEqual(second)
+  })
+
+  it('should fetch cached data when cached path is requested', async (): Promise<void> => {
+    const { result, waitForNextUpdate, rerender } = renderHook(
+      ({ initialValue }) =>
+        useFetch(
+          {
```

Comment:
you really like having the options be on their own line huh. I guess I should be more open minded about this.

Reply:
Can absolutely modify this to keep it consistent with the other tests. My bad